### DISCUSSION TITLE
Doubt about Twinning tutorial

### DISCUSSION THREAD

**Discussion by jucego333** [mtex-toolbox/mtex#975] [Ask Anything] [answered]
Dear Colleagues 

I hope you are fine. Recently, I have been used MText to define Twin boundaries in an fcc material. My confusion is coming in the part where we need to define the special orientation relationship. After prompt " round2Miller(mori_mean)", matlab shows the following result:

>> round2Miller(mori_mean)
 
(231) || (010)   [2-1-1] || [20-3]   error: 1.6°

After that I made Matlab to define a misorientation that makes a perfect fit between the lattice planes (231) and  (010) as is shows:

twinning = orientation.map(Miller(2,3,1,CS),Miller(2,3,1,CS),...
  Miller(0,1,0,CS),Miller(0,1,0,CS)) %%

Notwithstanding, I got the following :

crystal symmetry : Titanium cubic (m-3m)
  crystal symmetry : Titanium cubic (m-3m)
 
  Bunge Euler angles in degree
  phi1  Phi phi2 Inv.
     0    0    0    0

What was my mistake? Could anybody give me a guideline? Thanks in advance.

**Comment by ralfHielscher** [comment, marked as answer]
try 
```matlab
orientation.map(Miller({2 3 1},cs),Miller({0 1 0},cs),Miller({2 -1 -1},cs),Miller({2 0 -3},cs))
```
Ralf.

**Reply by jucego333** [reply]
Thank you so much!!!!!

**Comment by kanerto** [comment]
Hello there,

I am also trying to understand this code.
My calculation on Calcite grain boundaries gave me this:
(10-1-3) || (000-1)   [1-210] || [2-1-10]   error: 0.86°

So, by writing the following code:
twinning = orientation.map(Miller({1,0,-1,-3},CS('Calcite')) ,Miller({0,0,0,1},CS('Calcite')),...
  Miller({1,-2,1,0},CS('Calcite')), Miller({2,-1,-1,0},CS('Calcite')))

I got an error note: 
Index exceeds the number of array elements (3).

Neither a 3 element Miller index(h,k,l) or one less Miller definition helped and I got the same answer.
Also, I've noticed at the Magnesium example at the website the miller indices on the code were not as calculated by the "round2Miller(mori_mean)" function...

What should I do?

Thanks in advance!

**Reply by kilir** [reply]
Hi,
you may have to specify where the error occurs. As a result of the line you posted? If yo, which version of mtex are you using?
The way how a misorientation is determined should be unrelated to the error.
Cheers,
Rüdiger

**Reply by kanerto** [reply]
Yes, the error occurs when I'm activating the code I wrote.
I use mtex 5.6.0.

Thanks,
Tom

**Comment by ralfHielscher** [comment]
the code `CS('Calcite')` does not make any sense. You need either take calcite crystal symmetry from your EBSD data by
`CS = ebsd('calcite').CS` or define it directly.

Ralf.

**Reply by kanerto** [reply]
Ok now it works!

Thank you :)
